examples/deconv/train.cpp

```diff
@@ -37,17 +37,25 @@ void deconv_lanet(network<graph> &nn,
 
   // declare nodes
   input_layer i1(shape3d(32, 32, 1));
-  convolutional_layer<tan_h> c1(32, 32, 5, 1, 6);
+  convolutional_layer c1(32, 32, 5, 1, 6);
+  tanh_layer c1th(28, 28, 6);
   average_pooling_layer<tan_h> p1(28, 28, 6, 2);
-  deconvolutional_layer<tan_h> d1(14, 14, 5, 6, 16,
-                                  connection_table(tbl, 6, 16));
+  deconvolutional_layer d1(14, 14, 5, 6, 16, connection_table(tbl, 6, 16));
+  tanh_layer d1th(18, 18, 16);
```

Comment:
also activation layers I think that can be renamed at some point and remove the `_layer` part